### DISCUSSION TITLE
Redirect URL Error

### DISCUSSION THREAD

**Discussion by 0786ShivaniBansal** [JanssenProject/jans#5694] [Q&A] [answered]
Hi I'm getting invalid_request_redirect_uri while accessing the index test please suggest me a valid redirect URL this is the hostname server-j.us.corp.sennovate.com

```
**Error**{
    "error_description": "The redirect_uri in the Authorization Request does not match any of the Client's pre-registered redirect_uris. CorrelationId: 1b9fac57-75ed-4b2f-ad2c-89cf443a1854",
    "state": "MEOFZt8liyQ4FjssGAVt20emRSiUVJ",
    "error": "invalid_request_redirect_uri"
}
```

please view config.py

```
# REQUIRED
# Replace op_hostname
ISSUER = 'https://server-j.us.corp.sennovate.com:9090//oidc_callback'

# Tent redirect uri
REDIRECT_URIS = [
    'https://server-j.us.corp.sennovate.com/auth/callback'
]

# OPTIONAL: Use at your own risk

# Token authentication method can be
# client_secret_basic
# client_secret_post
# none
SERVER_TOKEN_AUTH_METHOD = 'client_secret_post'

# ACR VALUES
# Examples:
# ACR_VALUES = "agama"
# ACR_VALUES = 'simple_password_auth'
ACR_VALUES = None

# ADDITIONAL PARAMS TO CALL AUTHORIZE ENDPOINT, WITHOUT BASE64 ENCODING. USE DICT {'param': 'value'}
# ADDITIONAL_PARAMS = {'paramOne': 'valueOne', 'paramTwo': 'valueTwo'}
ADDITIONAL_PARAMS = None

# SYSTEM SETTINGS
# use with caution, unsecure requests, for development environments
SSL_VERIFY = False

# SCOPES
# Only scope "openid" is required for a pairwise identifier from the OP.
# OP can provision additional optional scopes as needed.
# SCOPE = 'openid email profile'
SCOPE = 'openid'
~                                     
```

**Comment by mzico** [comment]
@0786ShivaniBansal : can you please attach a screenshot of "accessing the index test" ?

**Comment by nynymike** [comment]
@0786ShivaniBansal -- Please use triple back quotes for long code snippets.

You are using Jans Tent? The `issuer` claim looks wrong--that should be the OpenID configuration endpoint URL.

**Comment by mmrraju** [comment, marked as answer]
Hello @0786ShivaniBansal 

If you use Jans tent, Please make sure
1. ISSUER = < your Authorization Server > like `https://server-j.us.corp.sennovate.com`
2. REDIRECT_URIS should be you RP client uri like for Jans-tent 'https://localhost:9090/oidc_callback'
3. Using Jans Tent dynamically client will be created. If want manual then you should create a client in your AS server then you must need to register the redirect uri, please check the attached file
![redirect_uri](https://github.com/JanssenProject/jans/assets/43112579/3e9f004c-4bad-4415-99f4-ec2882726e66)
.
4. Except register redirect URI you may get above error.

If need more assist please let me know.
